solidity/test/relay/LightRelayMaintainerProxy.test.ts

```diff
@@ -0,0 +1,532 @@
+import { concatenateHexStrings } from "../helpers/contract-test-helpers"
+import longHeaders from "./longHeaders.json"
+
+const { provider } = waffle
+
+const { createSnapshot, restoreSnapshot } = helpers.snapshot
+
+const ZERO_ADDRESS = ethers.constants.AddressZero
+
+const fixture = async () => {
+  await deployments.fixture()
+
+  const { deployer, governance } = await helpers.signers.getNamedSigners()
+  const [thirdParty, maintainer] = await helpers.signers.getUnnamedSigners()
+
+  const reimbursementPool: ReimbursementPool =
+    await helpers.contracts.getContract("ReimbursementPool")
+
+  const lightRelayMaintainerProxy: LightRelayMaintainerProxy =
+    await helpers.contracts.getContract("LightRelayMaintainerProxy")
+
+  const lightRelay: LightRelay = await helpers.contracts.getContract(
+    "LightRelay"
+  )
+
+  await lightRelay.connect(deployer).setAuthorizationStatus(true)
+
+  return {
+    deployer,
+    governance,
+    maintainer,
+    thirdParty,
+    reimbursementPool,
+    lightRelayMaintainerProxy,
+    lightRelay,
+  }
+}
+
+describe("LightRelayMaintainerProxy", () => {
+  let deployer: SignerWithAddress
+  let governance: SignerWithAddress
+  let maintainer: SignerWithAddress
+  let thirdParty: SignerWithAddress
+  let reimbursementPool: ReimbursementPool
+  let lightRelayMaintainerProxy: LightRelayMaintainerProxy
+  let lightRelay: LightRelay
+
+  before(async () => {
+    // eslint-disable-next-line @typescript-eslint/no-extra-semi
+    ;({
+      deployer,
+      governance,
+      maintainer,
+      thirdParty,
+      reimbursementPool,
+      lightRelayMaintainerProxy,
+      lightRelay,
+    } = await waffle.loadFixture(fixture))
+
+    await deployer.sendTransaction({
+      to: reimbursementPool.address,
+      value: ethers.utils.parseEther("100"),
+    })
+  })
+
+  describe("authorize", () => {
+    context("When called by non-owner", () => {
+      it("should revert", async () => {
+        await expect(
+          lightRelayMaintainerProxy
+            .connect(thirdParty)
+            .authorize(maintainer.address)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when called by the owner", () => {
+      context("When the maintainer is already authorized", () => {
+        before(async () => {
+          await createSnapshot()
+
+          // Authorize the maintainer to see if the next attempt reverts.
+          await lightRelayMaintainerProxy
+            .connect(governance)
+            .authorize(maintainer.address)
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should revert", async () => {
+          await expect(
+            lightRelayMaintainerProxy
+              .connect(governance)
+              .authorize(maintainer.address)
+          ).to.be.revertedWith("Maintainer is already authorized")
+        })
+      })
+
+      context("When the maintainer is not authorized yet", () => {
```

Comment:
```suggestion
      context("when the maintainer is not authorized yet", () => {
```

Reply:
Done in 462925b1bd07e05f87beb7c9fcba541d335c0be1.